admin-dev/themes/new-theme/js/components/router.js

```diff
@@ -43,8 +44,18 @@ const {$} = window;
  */
 export default class Router {
   constructor() {
+    const addRoutes = (customRoutes) => {
+      Object.assign(routes.routes, customRoutes);
+    };
+
+    EventEmitter.emit('beforeRoutes', addRoutes);
```

Comment:
After tests, we can't accept this change because the event is emitted before being able to plug on it :sweat_smile: